bitwarden_license/bit-web/src/app/secrets-manager/shared/secrets-list.component.html

```diff
@@ -171,6 +171,10 @@
     <i class="bwi bwi-fw bwi-refresh" aria-hidden="true"></i>
     <span>{{ "restoreSelected" | i18n }}</span>
   </button>
+  <button type="button" bitMenuItem *ngIf="!trash" (click)="bulkMoveToProject()">
```

Comment:
```suggestion
  <button type="button" bitMenuItem *ngIf="!trash && hasWriteAccessOnSelected$ | async" (click)="bulkMoveToProject()">
```

What about something like this? In secrets-list.componenet.ts add something like.

```
  protected hasWriteAccessOnSelected$ = this.selection.changed.pipe(
    map((_) => this.selectedHasWriteAccess()),
  );

  private selectedHasWriteAccess() {
    const selectedSecrets = this.secrets.filter((secret) =>
      this.selection.isSelected(secret.id),
    );
    if (selectedSecrets.some((secret) => secret.write)) {
      return true;
    }
    return false;
  }
```

This will hide the bulk move option if the user has read-only access to all the secrets selected.

This will prevent the user from making a bulk move with 0 secrets and follows SM's pattern of access restriction.